python/mxnet/metric.py

```diff
@@ -65,20 +116,36 @@ def get(self):
 class Accuracy(EvalMetric):
     """Calculate accuracy"""
 
-    def __init__(self):
+    def __init__(self, **kwargs):
         super(Accuracy, self).__init__('accuracy')
+        try:
+            self.top_k = kwargs['top_k']
+        except KeyError:
+            self.top_k = 1
+        assert(self.top_k > 0), 'Top k must be positive'
+        self.name += '-Top-%d' % self.top_k
 
     def update(self, labels, preds):
         check_label_shapes(labels, preds)
 
         for i in range(len(labels)):
-            pred_label = ndarray.argmax_channel(preds[i]).asnumpy().astype('int32')
+            if self.top_k > 1:
+                pred_label = numpy.argsort(preds[i].asnumpy().astype('float32'), axis=1)
+            else:
+                pred_label = ndarray.argmax_channel(preds[i]).asnumpy().astype('int32')
             label = labels[i].asnumpy().astype('int32')
 
             check_label_shapes(label, pred_label)
-
-            self.sum_metric += (pred_label.flat == label.flat).sum()
-            self.num_inst += len(pred_label.flat)
+            num_samples = pred_label.shape[0]
+            num_dims = len(pred_label.shape)
+            if num_dims == 1:
```

Comment:
check top_k here instead of num_dims.
pred_label can have multiple dimensions for multi label prediction tasks.